packages/cdktf/lib/terraform-move-addresses.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { IConstruct } from "constructs";
+import { TerraformResource } from "./terraform-resource";
+import { TerraformStack } from "./terraform-stack";
+import { App } from "./app";
+
+/**
+ *
+ */
+export class TerraformResourceAddressMap {
+  private _terraformAddressMap: Map<string, TerraformResource> = new Map<
+    string,
+    TerraformResource
+  >();
+
+  // what do we need?
+  /**
+   * map for a tag -> resource path !!!
+   * function to add to the resource path at a tag
+   * function to return the specificed resource address at a certain tag
+   */
+
+  public add(resource: TerraformResource, tag: string): void {
+    this._terraformAddressMap.set(tag, resource);
+  }
+
+  public getResourceAddress(tag: string): TerraformResource | undefined {
+    return this._terraformAddressMap.get(tag);
+  }
+
+  public static parentStackAddressMap(scope: IConstruct) {
+    // SO, in a nested construct, the stack is accessable at scope.node.scope ACTUALLY NOT TRUE
+    // -- only thought that was the case as scope.node.scope just happened to be the stack
+    // But, if you are already at the root level it is undefined
+    // what if we keep trying to access the scope until it is undefined
+
+    // if scope.scope === undefined we are at the stack level
+
+    let currAbstractionLevel: IConstruct | undefined = scope;
+    let oneHigher: IConstruct | undefined = currAbstractionLevel.node.scope;
+    while (!App.isApp(oneHigher)) {
+      currAbstractionLevel = currAbstractionLevel?.node.scope;
+      oneHigher = currAbstractionLevel?.node.scope;
+    }
```

Comment:
There is also `App.of(scope)` to just get the app 😉

Reply:
That said, `TerraformStack.of(scope)` might be what could simplify this entire method even more 😉

Reply:
Bless you for informing me of this 😌